src/Magnum/Platform/WindowlessEglApplication.cpp

```diff
@@ -157,12 +157,55 @@ WindowlessEglContext::WindowlessEglContext(const Configuration& configuration, G
                 Debug{} << "Platform::WindowlessEglApplication: found" << count << "EGL devices, choosing device" << configuration.device();
             }
 
-            /* Assuming the same thing won't suddenly start failing when called
-               the second time */
-            Containers::Array<EGLDeviceEXT> devices{configuration.device() + 1};
-            CORRADE_INTERNAL_ASSERT_OUTPUT(eglQueryDevices(configuration.device() + 1, devices, &count));
+            UnsignedInt selectedDevice;
+            Containers::Array<EGLDeviceEXT> devices;
+
+            if (configuration.deviceIsCuda()) {
```

Comment:
Following the API changes, this would be 

```suggestion
            if (configuration.cudaDevice() != ~UnsignedInt{}) {
```